Last-Update: 2013-07-24
Forwarded: not-needed
Origin: upstream, svn://svn.zabbix.com/branches/dev@37228
Bug-Zabbix: https://support.zabbix.com/browse/ZBX-4986
Author: Dmitry Smirnov <onlyjob@member.fsf.org>
Description: Changed getUrl method for returning relative url's.
 Fix for columns sorting and pagination in frontend-php
 Problem is manifesting with nginx/php-fpm or with Apache/mod_proxy.

--- a/frontends/php/include/classes/class.curl.php
+++ b/frontends/php/include/classes/class.curl.php
@@ -201,17 +201,28 @@ class Curl {
 		}
 		$this->formatQuery();
 	}
-
-	public function getUrl() {
+	/**
+	 * Return relative url.
+	 *
+	 * @param boolean $absoluteUrl   required absolute url
+	 *
+	 * @return string
+	 */
+	public function getUrl($absoluteUrl = false) {
 		$this->formatQuery();
 
-		$url = $this->protocol ? $this->protocol.'://' : '';
-		$url .= $this->username ? $this->username : '';
-		$url .= $this->password ? ':'.$this->password : '';
-		$url .= $this->username || $this->password ? '@' : '';
-		$url .= $this->host ? $this->host : '';
-		$url .= $this->port ? ':'.$this->port : '';
-		$url .= $this->path ? $this->path : '';
+		$url = '';
+		if($absoluteUrl) {
+			$url .= $this->protocol ? $this->protocol.'://' : '';
+			$url .= $this->username ? $this->username : '';
+			$url .= $this->password ? ':'.$this->password : '';
+			$url .= $this->username || $this->password ? '@' : '';
+			$url .= $this->host ? $this->host : '';
+			$url .= $this->port ? ':'.$this->port : '';
+			$url .= $this->path ? $this->path : '';
+		} else {
+			$url .= $this->path ? basename($this->path) : '';
+		}
 		$url .= $this->query ? '?'.$this->query : '';
 		$url .= $this->reference ? '#'.urlencode($this->reference) : '';
 		return $url;
